### PR TITLE
add actors_model_example

### PR DESCRIPTION
**Pull request by azimgivron**
Let me know if there are changes to make. 
The code is the same as in the normal scaffold/python with the src/my_module.py that was updated. 
However, I only added the '&& pip install thespian==3.10.3' in the Dockerfile but it does not seem to be rebuilt when running the build script. Can you tell me what I should add to solve this ?

### CODE REVIEW

File: scaffolds_actor_model/python/python-env [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+VERSION="1"
+TAG="$IMAGE:$VERSION"
+
+docker images | grep -q "$IMAGE[ ]*$VERSION" || docker build -t $TAG .
+
+docker run --rm -i ${ENABLE_TTY} \
+  -u $(id -u):$(id -g) \
+  -v $(pwd):/tmp/project:rw \
+  -w /tmp/project \
+  $TAG $@
```
Comment (jbruggem) [inline scaffolds_actor_model/python/python-env:16]: Nit-picking: there's a new line missing at the end of the file

File: scaffolds_actor_model/python/src/my_module.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+     #create actors
+     actor1 = actor_sys.createActor(MyClass)
+     actor2 = actor_sys.createActor(MyClass)
+
+     #send address of actor2 to actor1  
+     actor_sys.tell(actor1, {"msg_type": "receive_address", "address": actor2})
+
+     #tell actor1 to send a message its neighbour
+     actor_sys.tell(actor1, {"msg_type": "send_msg", "content": "hello world"})
```
Comment (jbruggem) [inline scaffolds_actor_model/python/src/my_module.py:28]: nit-picking: missing empty line at EOF

File: scaffolds_actor_model/python/src/my_module.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import thespian.actors as ta
+
+class MyClass(ta.Actor):
+     def __init__(self, **kwargs):
+          super().__init__(**kwargs)
+          self.neighbour = None
+
+     def receiveMessage(self, msg, sender):
+          if msg["msg_type"] == "receive_address":
+               self.neighbour = msg["address"]
```
Comment (jbruggem) [inline scaffolds_actor_model/python/src/my_module.py:10]: The use of *neighbour* makes me think that you had a particular use case (our experiment with a concurrent GoL), but this example should be use-case agnostic. Maybe renaming the variable is enough ?

WDYT ?

File: scaffolds_actor_model/python/src/test/test_my_module.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import pytest
+from my_module import GameOfLife
```
Comment (jbruggem) [inline scaffolds_actor_model/python/src/test/test_my_module.py:2]: This example should be agnostic, so shouldn't mention any particular use case.

Besides, I don't see any `GameOfLife` in `my_module`.

Could you make sure that references to GoL are removed, and that the tests of the scaffold pass (there should only be one or two, as an example of how to write tests)

**Review by jbruggem — CHANGES_REQUESTED** [review]
Great start !

Two things:
- some clean up to do.
- it missing an example of how to get an actor's address based on its name. Is that possible with thespian ?

thanks for your contribution !

**Comment by azimgivron** [comment]
> Great start !
> 
> Two things:
> 
> * some clean up to do.
> * it missing an example of how to get an actor's address based on its name. Is that possible with thespian ?
> 
> thanks for your contribution !

I made the requested changes, let me know if there is something else to modify. 
The reason I left the address passing in as initialization message is because to get the actors by name, you need to call the actor system and I am wondering whether it is not a bottleneck since it needs to be centralized. However, both options are in the example.

**Pull request merged**